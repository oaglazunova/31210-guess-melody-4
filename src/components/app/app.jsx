import React from 'react';
import {WelcomeScreen} from '../welcomeScreen/welcomeScreen.jsx';
import PropTypes from "prop-types";

export const App = (props) => {
  const {errorCount} = props;

  return <WelcomeScreen errorCount={errorCount} />;
};

App.propTypes = {
  errorCount: PropTypes.number,
};
