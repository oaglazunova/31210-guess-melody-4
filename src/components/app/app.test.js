import React from 'react';
import renderer from 'react-test-renderer';
import {describe, expect, it} from '@jest/globals';
import {App} from './app.jsx';

const testData = {
  errorCount: 3
};

describe(`Content rendered`, () => {
  // Step 1: Snapshot test
  it(`renders correctly`, () => {
    const tree = renderer.create(
        <App errorCount={testData.errorCount}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
