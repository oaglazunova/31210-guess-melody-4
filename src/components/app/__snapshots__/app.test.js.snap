// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content rendered renders correctly 1`] = `
<main
  className="app"
>
  <svg
    style={
      Object {
        "left": "-1200em",
        "position": "absolute",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <filter
      id="blur"
    >
      <feGaussianBlur
        in="SourceGraphic"
        stdDeviation="5"
      />
      <feOffset
        dx="0"
        dy="0"
      />
      <feMerge>
        <feMergeNode />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </svg>
  <section
    className="welcome"
  >
    <div
      className="welcome__logo"
    >
      <img
        alt="Угадай мелодию"
        height="83"
        src="img/melody-logo.png"
        width="186"
      />
    </div>
    <button
      className="welcome__button"
      onClick={[Function]}
    >
      <span
        className="visually-hidden"
      >
        Начать игру
      </span>
    </button>
    <h2
      className="welcome__rules-title"
    >
      Правила игры
    </h2>
    <p
      className="welcome__text"
    >
      Правила просты:
    </p>
    <ul
      className="welcome__rules-list"
    >
      <li>
        За 10 минут нужно ответить на все вопросы.
      </li>
      <li>
        Можно допустить 
        3
         ошибки.
      </li>
    </ul>
    <p
      className="welcome__text"
    >
      Удачи!
    </p>
  </section>
</main>
`;
